mono/mini/debugger-agent.c

```diff
@@ -837,6 +837,30 @@ static void
 register_socket_transport (void);
 #endif
 
+#ifndef IL2CPP_MONO_DEBUGGER
+static MonoAssembly* mono_domain_get_assemblies_iter(MonoDomain *domain, void* *iter)
+{
+	if (!iter)
+		return NULL;
+
+	if (!*iter)
```

Comment:
Could be simplified to
```
if (*iter)
	*iter = ((GList*)(*iter))->next;
else
	*iter = domain->domain_assemblies;


if (*iter)
	return ((GList*)(*iter))->data;
else
	return NULL;
```

Reply:
I will look at simplifying this method in the next PR.